src/audio/src/src_hifi4.c

```diff
@@ -231,27 +231,24 @@ static inline void fir_filter(ae_f32 *rp, const void *cp, ae_f32 *wp0,
 			 * Q1.23 values.
 			 */
 			data2 = AE_SELP24_LL(d0, d1);
-			AE_MULAAFP24S_HH_LL(a0, data2, coef2);
+			AE_MULAAFD32S_HH_LL(a0, data2, coef2);
 			data2 = AE_SELP24_HH(d0, d1);
-			AE_MULAAFP24S_HH_LL(a1, data2, coef2);
+			AE_MULAAFD32S_HH_LL(a1, data2, coef2);
 
 			/* Repeat for next two taps */
 			AE_L32X2F24_IP(coef2, coefp, sizeof(ae_f24x2));
 			AE_L32X2F24_XC(d0, dp, inc); /* r2, l2 */
 			AE_L32X2F24_XC(d1, dp, inc); /* r3, l3 */
 			data2 = AE_SELP24_LL(d0, d1);
-			AE_MULAAFP24S_HH_LL(a0, data2, coef2);
+			AE_MULAAFD32S_HH_LL(a0, data2, coef2);
 			data2 = AE_SELP24_HH(d0, d1);
-			AE_MULAAFP24S_HH_LL(a1, data2, coef2);
+			AE_MULAAFD32S_HH_LL(a1, data2, coef2);
 		}
 
 		/* Scale FIR output with right shifts, round/saturate
 		 * to Q1.31, and store 32 bit output.
 		 */
-		AE_S32_L_XP(AE_ROUND32F48SSYM(AE_SRAA64(a0, shift)), wp,
-			    sizeof(int32_t));
-		AE_S32_L_XP(AE_ROUND32F48SSYM(AE_SRAA64(a1, shift)), wp,
-			    sizeof(int32_t));
+		AE_S24X2RA64S_IP(a0, a1, (ae_f24x2 *)wp);
```

Comment:
This is wrong format, not Q1.31, but MSB aligned Q1.23. The original code supports ALSA formats S32_LE and S24_LE (LSB aligned). Please don't break it but add new features if you need MSB aligned Q1.23.